update/Dockerfile

```diff
@@ -50,6 +50,12 @@ RUN --mount=type=secret,id=STEAM_USERNAME \
         linux32/libvstdlib_s.so \
         linux32/steamclient.so \
         linux32/steamconsole.so \
-        update_hosts_cached.vdf \
-        /root/.steam \
-        /root/Steam
+        update_hosts_cached.vdf; \
+    cd /root && rm -rf \
+        .steam; \
+    if [ "$APPID" = 90 ]; then \
+        # Keep `/root/Steam/appcache` to prevent `steamcmd` from re-downloading `hlds` games from scratch whenever the game is updated
+        rm -rf $( find Steam -maxdepth 1 | grep -vE '^Steam$|^Steam/appcache' ); \
+    else \
+        rm -rf Steam; \
+    fi
```

Comment:
```suggestion
    if [ "$APPID" = 90 ]; then \
        # Keep `/root/Steam/appcache` to prevent `steamcmd` from re-downloading `hlds` games from scratch whenever the game is updated
        rm -rf \
          /root/Steam/config \
          /root/Steam/depotcache \
          /root/Steam/logs \
          /root/Steam/steamapps \
          /root/Steam/userdata; \
    else \
        rm -rf /root/Steam; \
    fi;
```

Reply:
There are other files well. Removing by exclusion works better imo

Reply:
perhaps we could `ls` the directory and ensure there's nothing else in there for `hlds`?

Reply:
```suggestion
    rm -rf /root/.steam
    if [ "$APPID" = 90 ]; then \
        # Keep `/root/Steam/appcache` to prevent `steamcmd` from re-downloading `hlds` games from scratch whenever the game is updated
        rm -rf $( find /root/Steam -mindepth 1 -maxdepth 1 | grep -v ^/root/Steam/appcache )
    else \
        rm -rf /root/Steam; \
    fi;
```

is this better?

Reply:
at least for `hlds-cstrike`  there's a binary file that's named `Steam/ssfn*`:

```
#9 85.53 removed 'Steam/ssfn4078290948139867735'
```

doesn't appear to exist for `srcds-hl2mp`

Reply:
yes

Reply:
`rm -rf /root/Steam` is good for `srcds`. guess we could just add `/root/Steam/ssfn*` to the `hlds` list

Reply:
does [it](https://github.com/startersclan/docker-sourceservers/pull/113#discussion_r1125615362) work? find /root/Steam would include `/root/Steam` which will entirely be removed

Reply:
think its better to push the change so it can be fetched

Reply:
changes would be reviewable as well

Reply:
commit the suggestion